Marlin/src/lcd/tft/canvas.cpp

```diff
@@ -139,8 +150,25 @@ void Canvas::addImage(int16_t x, int16_t y, MarlinImage image, uint16_t *colors)
               dsty = y, dstx = x;                   // Destination line / column index
 
       uint16_t color = 0;                           // Persist the last fetched color value
+      if (rle_state.has_rle_state) {                                // do we have rle position data
+        dstx = rle_state.dstx;                      // restore dstx
+        dsty = rle_state.dsty;                      // restore dstx
+        srcx = rle_state.srcx;                      // restore dstx
+        srcy = rle_state.srcy;                      // restore dstx
+        bytedata = (uint8_t *)images[image].data + rle_state.rle_offset;  // restart decode from here instead of teh start of data
```

Comment:
```suggestion
        bytedata = (uint8_t *)images[image].data + rle_state.rle_offset;  // restart decode from here instead of the start of data
```

Reply:
fixed,